### PR TITLE
[6.15.z] Replace old method of registration with global registration in robotlelo

### PR DESCRIPTION
**Pull request by shweta83**
…elo (#14034)

### CODE REVIEW

**Review by jyejare — COMMENTED** [review]
Looks good to me from framework changes perspective and integration with tests( I did not go in each test though).

So, lets run some PRT.

File: tests/foreman/api/test_registration.py [hunk 1]
```diff
@@ -70,17 +68,14 @@ def test_host_registration_end_to_end(
     module_target_sat.api.SmartProxy(id=nc.id, organization=[org]).update(['organization'])
     module_target_sat.api.SmartProxy(id=nc.id, location=[module_location]).update(['location'])
 
-    command = module_target_sat.api.RegistrationCommand(
-        smart_proxy=nc,
+    result = rhel_contenthost.api_register(
+        nc,
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_registration.py:72]: Could you cherrypick changes from https://github.com/SatelliteQE/robottelo/pull/15306 ?

Comment (shweta83) [reply tests/foreman/api/test_registration.py]: Done

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_reporttemplates.py -k 'test_positive_generate_entitlements_report'

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_reporttemplates.py -k test_positive_generate_entitlements_report

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_reporttemplates.py -k test_positive_generate_entitlements_report

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7360
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_reporttemplates.py -k test_positive_generate_entitlements_report --external-logging
Test Result : ========== 1 passed, 36 deselected, 49 warnings in 960.42s (0:16:00) ===========
```

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**